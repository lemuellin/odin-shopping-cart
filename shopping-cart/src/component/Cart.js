import React from 'react';
import '../style/cart.css';

import Echeveria from '../asset/Echeveria';
import Cactus from '../asset/Cactus';
import Aloe from '../asset/Aloe';


const Cart = (props) => {
    let countEcheveria = 5; // 0-4
    let countCactus = 11; // 5-15
    let countAloe = 4; // 16-19
    let sum = 0;

    const checkout = () => {
        alert('Did you really just tried to buy these fake plants?');
    }

    return(    
        <div className='cartList'>
            <div className='cartCard'>
                <h2 className='itemsTitle'>YOUR CART</h2>
                <div>
                    {props.content.map((item,index) => {
                        if (item !== 0 && index < countEcheveria){
                            sum += item * Echeveria[index].price;
                            return(
                                <div key={index} id={index} className='cartItem'>
                                    <img src={Echeveria[index].img} alt='Succulent Photos' className='shopItemPhoto'/>
                                    <div className='cartItemAmount'>
                                        <h3>{Echeveria[index].name}</h3>
                                        <h4>{'$ ' + Echeveria[index].price}</h4>
                                        <form className='amount'>    
                                            <button type='button' className='amountBtn' onClick={props.minusButton}>-</button>
                                            <input type='number' className='amountInput' value={item} onChange={props.inputField}/>
                                            <button type='button' className='amountBtn' onClick={props.plusButton}>+</button>
                                        </form>
                                    </div>
                                    <button className='removeButton' onClick={props.removeButton}>X</button>
                                </div>
                            );
                        }else if(item !== 0 && index < (countEcheveria + countCactus)){
                            sum += item * Cactus[index - 5].price;
                            return(
                                <div key={index} id={index} className='cartItem'>
                                    <img src={Cactus[index - 5].img} alt='Succulent Photos' className='shopItemPhoto'/>
                                    <div className='cartItemAmount'>
                                        <h3>{Cactus[index - 5].name}</h3>
                                        <h4>{'$ ' + Cactus[index - 5].price}</h4>
                                        <form className='amount'>    
                                            <button type='button' className='amountBtn' onClick={props.minusButton}>-</button>
                                            <input type='number' className='amountInput' value={item} onChange={props.inputField}/>
                                            <button type='button' className='amountBtn' onClick={props.plusButton}>+</button>
                                        </form>
                                    </div>
                                    <button className='removeButton' onClick={props.removeButton}>X</button>
                                </div>
                            );
                        }else if(item !== 0){
                            sum += item * Aloe[index - 16].price;
                            return(
                                <div key={index} id={index} className='cartItem'>
                                    <img src={Aloe[index - 16].img} alt='Succulent Photos' className='shopItemPhoto'/>
                                    <div className='cartItemAmount'>
                                        <h3>{Aloe[index - 16].name}</h3>
                                        <h4>{'$ ' + Aloe[index - 16].price}</h4>
                                        <form className='amount'>
                                            <button type='button' className='amountBtn' onClick={props.minusButton}>-</button>
                                            <input type='number' className='amountInput' value={item} onChange={props.inputField}/>
                                            <button type='button' className='amountBtn' onClick={props.plusButton}>+</button>
                                        </form>
                                    </div>
                                    <button className='removeButton' onClick={props.removeButton}>X</button>
                                </div>
                            );
                        }else{
                            return null;
                        }
                    })}                
                </div>
                <div className='total'>
                    <h4>Subtotal: </h4>
                    <h4>{'$ ' + sum.toFixed(2)}</h4>
                    <h4>Shipping:</h4>
                    <h4>free</h4>
                    <h4>Tax:</h4>
                    <h4>{'$ ' + (sum * 0.12).toFixed(2)}</h4>
                    <h3>Grand Total: </h3>
                    <h3>{'$ ' + (sum * 1.1).toFixed(2)}</h3>
                </div>
                <div className='buttons'>
                    <button onClick={props.showCart} className='backToShop'>Back To Shop</button>
                    <button onClick={checkout} className='checkout'>Checkout</button>
                </div>
            </div>
        </div>
    )
}

export default Cart;