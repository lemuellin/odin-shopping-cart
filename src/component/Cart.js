import React from 'react';
import '../style/cart.css';

import Echeveria from '../asset/Echeveria';
import Cactus from '../asset/Cactus';
import Aloe from '../asset/Aloe';

import CartItem from './CartItem';


const Cart = (props) => {
    let startPosEcheveria = 0; // 0-4
    let startPosCactus = 5; // 5-15
    let startPosAloe = 16; // 16-19
    let sum = 0;

    const checkout = () => {
        alert('Did you really just tried to buy these fake plants?');
    }

    return(    
        <div className='cartList'>
            <div className='cartCard'>
                <h2 className='itemsTitle'>YOUR CART</h2>
                <div>
                    {props.content.map((item,index) => {
                        if (item !== 0 && index < startPosCactus){
                            sum += item * Echeveria[index].price;
                            return(
                                CartItem(props, item, index, Echeveria, startPosEcheveria)
                            );
                        }else if(item !== 0 && index < (startPosAloe)){
                            sum += item * Cactus[index - 5].price;
                            return(
                                CartItem(props, item, index, Cactus, startPosCactus)
                            );
                        }else if(item !== 0){
                            sum += item * Aloe[index - 16].price;
                            return(
                                CartItem(props, item, index, Aloe, startPosAloe)
                            );
                        }else{
                            return null;
                        }
                    })}                
                </div>
                <div className='total'>
                    <h4>Subtotal: </h4>
                    <h4>{'$ ' + sum.toFixed(2)}</h4>
                    <h4>Shipping:</h4>
                    <h4>free</h4>
                    <h4>Tax:</h4>
                    <h4>{'$ ' + (sum * 0.12).toFixed(2)}</h4>
                    <h3>Grand Total: </h3>
                    <h3 data-testid='gTotal'>{'$ ' + (sum * 1.1).toFixed(2)}</h3>
                </div>
                <div className='buttons'>
                    <button onClick={props.showCart} className='backToShop'>Back To Shop</button>
                    <button onClick={checkout} className='checkout'>Checkout</button>
                </div>
            </div>
        </div>
    )
}

export default Cart;