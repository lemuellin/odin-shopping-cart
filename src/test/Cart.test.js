import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from '@testing-library/user-event';
import Cart from "../component/Cart";

describe("Cart Component", () => {
    it("Match Snapshot", () => {
        const cart = [];
        const { container } = render(
            <Cart 
                content={cart} 
                inputField={() => console.log('inputField')} 
                minusButton={() => console.log('minusButton')}  
                plusButton={() => console.log('plusButton')} 
                removeButton={() => console.log('removeButton')} 
                showCart={() => console.log('showCart')} 
        />);

        expect(container).toMatchSnapshot();
    });
    it("Calculate total fee", () => {
        const cart = [1,5,3,0,0,0,0,77,8,9];
        render(
            <Cart 
                content={cart} 
                inputField={() => console.log('inputField')} 
                minusButton={() => console.log('minusButton')}  
                plusButton={() => console.log('plusButton')} 
                removeButton={() => console.log('removeButton')} 
                showCart={() => console.log('showCart')} 
        />);

        const gTotal = screen.getByTestId('gTotal');

        expect(gTotal.textContent).toBe('$ 613.80');
    });
});