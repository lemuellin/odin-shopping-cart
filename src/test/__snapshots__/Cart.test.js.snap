// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Component Match Snapshot 1`] = `
<div>
  <div
    class="cartList"
  >
    <div
      class="cartCard"
    >
      <h2
        class="itemsTitle"
      >
        YOUR CART
      </h2>
      <div />
      <div
        class="total"
      >
        <h4>
          Subtotal: 
        </h4>
        <h4>
          $ 0.00
        </h4>
        <h4>
          Shipping:
        </h4>
        <h4>
          free
        </h4>
        <h4>
          Tax:
        </h4>
        <h4>
          $ 0.00
        </h4>
        <h3>
          Grand Total: 
        </h3>
        <h3
          data-testid="gTotal"
        >
          $ 0.00
        </h3>
      </div>
      <div
        class="buttons"
      >
        <button
          class="backToShop"
        >
          Back To Shop
        </button>
        <button
          class="checkout"
        >
          Checkout
        </button>
      </div>
    </div>
  </div>
</div>
`;
